ospf6d/ospf6_gr_helper.c

```diff
@@ -155,7 +155,9 @@ static int ospf6_extract_grace_lsa_fields(struct ospf6_lsa *lsa,
 
 	lsah = (struct ospf6_lsa_header *)lsa->header;
 
-	length = ntohs(lsah->length) - OSPF6_LSA_HEADER_SIZE;
+	length = (ntohs(lsah->length) > OSPF6_LSA_HEADER_SIZE)
+		? (ntohs(lsah->length) - OSPF6_LSA_HEADER_SIZE)
+		: 0;
```

Comment:
we need to compare length against remaining packet length too right?  What happens if the length read in is 50 but packet length left is 40.  50 is still greater than OSPF6_LSA_HEADER_SIZE right>

Reply:
It‘s a check to avoid “length” flip. I think it will fix coverity warning.

Reply:
Do you think the check in `ospf6_message.c:ospf6_lsupdate_recv()` is doing enough?  It looks like the lsa length has been validated, but sort of implicitly in the `for` loop there. The message-level functions do more explicit validations; it might be clearer if there was an explicit sanity-check during the per-lsa loop.